client/src/components/User/PreferencePage.vue

```diff
@@ -0,0 +1,54 @@
+<script setup lang="ts">
+import { faLongArrowAltLeft } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
+import { BButton } from "bootstrap-vue";
+import { computed } from "vue";
+
+import Heading from "@/components/Common/Heading.vue";
+
+const props = defineProps<{
+    heading: string;
+    icon: any;
```

Comment:
If you want to be more accurate here with the icon type you can `import { type IconDefinition } from "@fortawesome/fontawesome-svg-core";` and "force" it to be an `IconDefinition` instead of a string. If we start doing this eventually we can get rid of the "string" icon representation.

Reply:
I'd rather do this in a separate PR when replacing the icons